pkg/ds/mutexmap/mutexmap.go

```diff
@@ -0,0 +1,31 @@
+package mutexmap
+
+import "sync"
+
+type MutexMap[K comparable, V any] struct {
+	data map[K]V
+	l    sync.RWMutex
+}
+
+func NewMutexMap[K comparable, V any]() MutexMap[K, V] {
+	return MutexMap[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+func (m *MutexMap[K, V]) Store(k K, v V) {
+	m.l.Lock()
+	m.data[k] = v
+	m.l.Unlock()
+}
+
+func (m *MutexMap[K, V]) Load(k K) (v V, ok bool) {
+	m.l.RLock()
+	defer m.l.RUnlock()
+	v, ok = m.data[k]
```

Comment:
can be `return v, ok`

Reply:
updated in https://github.com/KyberNetwork/tradinglib/pull/4/commits/d5bc0e5289d25321cdc0579b8d914558b8d5a7ca